File: MEMEFLOW_SETTINGS_AUDIT_V7/rollback.mjs
import fs from 'node:fs';
import path from 'node:path';
const cwd=process.cwd();
const appDir=fs.existsSync(path.join(cwd,'memeflow-app'))?path.join(cwd,'memeflow-app'):cwd;
const rel=['src/settings.mjs','src/evaluate.mjs','src/paper-engine.mjs','src/store.mjs','app-server.mjs','index.html'];
for(const r of rel){const p=path.join(appDir,r),b=p+'.before-settings-audit-v7';if(!fs.existsSync(b)){console.error('ABORT: missing backup '+b);process.exit(1)}}
for(const r of rel){const p=path.join(appDir,r);fs.copyFileSync(p+'.before-settings-audit-v7',p);console.log('Restored:',p)}
console.log('V7 rollback complete.');

File: MEMEFLOW_OPENAI_PATCH_V1/rollback.py
from pathlib import Path
import shutil,sys
ROOT=Path.cwd();m=ROOT/".memeflow-openai-last-backup"
if not m.exists():print("No backup marker found.");sys.exit(1)
b=Path(m.read_text().strip())
if not b.exists():print("Backup folder not found:",b);sys.exit(2)
for rel in ["app-server.mjs","index.html",".env.example","src/store.mjs"]:
    s=b/rel
    if s.exists():
        d=ROOT/rel;d.parent.mkdir(parents=True,exist_ok=True);shutil.copy2(s,d)
p=ROOT/"src"/"openai-intelligence.mjs"
if p.exists():p.unlink()
print("ROLLBACK COMPLETE:",b)
